tests/test_GeoSpace.py

```diff
@@ -143,6 +143,10 @@ def test_get_agents_as_GeoDataFrame(self):
             for agent in self.agents
         ]
         agents_gdf = gpd.GeoDataFrame.from_records(agents_list, index="unique_id")
+        # workaround for geometry column not being set in `from_records`
+        # see https://github.com/geopandas/geopandas/issues/3152
+        # may be removed when the issue is resolved
+        agents_gdf.set_geometry("geometry", inplace=True)
```

Comment:
Does the fact that this has to be present in test code means users will have to add this line in their code?

Reply:
Thanks for responding! Appreciate it. No users don't need to add this. Instead they will use the `geo_space.get_agents_as_GeoDataFrame()` method to extract geoagents as a geodataframe.

This is a test method to check if `geo_space.get_agents_as_GeoDataFrame()` returns a geodataframe of geoagents as expected.